File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int index = 0, end = nums.size();
        while(index < end){
            if(nums[index] == val){
                nums[index] = nums[end-1];
                end--;
            }
            else{
                index++;
            }
        }
        
        return index;
    }
};

File: 0027-remove-element/0027-remove-element.java
class Solution {
    public void swap(int[] nums, int i, int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
    public int removeElement(int[] nums, int val) {
        int r = nums.length-1;
        int n = nums.length;
        int i = 0;

        while(i<=r){
            if(nums[i]==val){
                swap(nums, i, r);
                r--;
            }else{
                i++;
            }
        }
        return i;
    }
}